### ISSUE TITLE
Warning: ReactDOM.render is no longer supported in React 18 when using ReactTestUtils.renderIntoDocument

### ISSUE THREAD

**Issue by raaaahman** [facebook/react#25446]
React version: 18.2.0
ReactDOM version: 18.2.0

## Steps To Reproduce

1. Setup Jest
2. Create a test file which calls to the `ReactTestUtils.renderIntoDocument`method
3. Run jest

Example code:

```javascript
test('component should render', () => {
  const container = renderIntoDocument(<MyComponent/>)

  expect(findRenderedComponentWithType(container, MyComponent)).not.toBeNull()
})
```
## The current behavior

When using `ReactTestUtils.renderIntoDocument`in React 18, I have the following warning in the console, that make the test fail:

> console.error
>      Warning: ReactDOM.render is no longer supported in React 18. Use createRoot instead. Until you switch to the new API, your app will behave as if it's running React 17. Learn more: https://reactjs.org/link/switch-to-createroot

It seems to come from this line in the source code: https://github.com/facebook/react/blob/7b25b961df878109a2b3810f33815249cae37ecc/packages/react-dom/src/test-utils/ReactTestUtils.js#L128

## The expected behavior

I expect the `ReactTestUtils.renderIntoDocument` method not to trigger a `console.error` and to return a DOM Node.

I know most of the people are probably using `@testing-library/react` these days, should we expect the React Test Utilities to be deprecated soon?

Thank you for reading.

**Comment by gaearon** [comment]
Yeah, TestUtils are essentially on life support. We try not to break them but they're frozen in functionality. `renderIntoDocument` doesn't do anything useful — it's an alias for `ReactDOM.render` into a new div (despite the naming). So I suggest to use `createRoot` directly or use Testing Library.

**Comment by raaaahman** [comment]
Gotcha! Thank you for the quick reply.

**Comment by YLeventhal** [comment]
> So I suggest to use createRoot directly

@gaearon  but the legacy` ReactDOM.render` returned `React$Component<any, any> | PublicInstance | null ` while the new api returns void from render ?

**Comment by eryue0220** [comment]
Is there any progress on this issue?
